classes/cliTool/CliTool.inc.php

```diff
@@ -81,4 +89,56 @@ function __construct($argv = array()) {
 	function usage() {
 	}
 
+	private function checkArgsForUsername() {
+		$usernameKeyPos = array_search('--user_name', $this->argv);
+		if ($usernameKeyPos) {
+			$usernamePos = $usernameKeyPos + 1;
+			if (count($this->argv) >= $usernamePos + 1) {
+				$this->username = $this->argv[$usernamePos];
+
+				unset($this->argv[$usernamePos]);
+			}
+
+			unset($this->argv[$usernameKeyPos]);
+		}
+
+		$userDao = DAORegistry::getDAO('UserDAO'); /** @var $userDao UserDAO */
+
+		if ($this->username) {
+
+			$user = $userDao->getByUsername($this->username);
+
+			if ($user) {
+				$this->user = $user;
+			}
+		}
+
+		if (!$this->user) {
+			$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /** @var $userGroupDao UserGroupDAO */
+			$adminGroups = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_SITE_ADMIN);
+
+			if (count($adminGroups)) {
+				$groupUsers = $userGroupDao->getUsersById($adminGroups[0])->toArray();
+
+				$this->setUser($groupUsers[0]);
```

Comment:
Maybe double-check if the user exists and `exit(0)` if not?

Reply:
Fixed